lib/kafka/ssl_socket_with_timeout.rb

```diff
@@ -163,6 +163,10 @@ def closed?
       @tcp_socket.closed? || @ssl_socket.closed?
     end
 
+    def to_io
```

Comment:
I am not sure this works. Just because the backing tcp socket is ready doesn't mean the ssl socket is ready. It probably works well enough in practice, but we would need to test that it does.